pkg/dashboard/functiontemplates/zipfunctiontemplatefetcher.go

```diff
@@ -0,0 +1,149 @@
+	var functionsFileContents map[string]*FunctionTemplateFileContents
+
+	zftf.logger.DebugWith("Getting the zip file from the given address", "fileAddress", zftf.fileAddress)
+	response, err := http.Get(zftf.fileAddress)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get zip file")
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read response body")
+	}
+
+	zipReader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create zip reader")
+	}
+
+	functionsFileContents = zftf.parseFiles(zipReader)
+
+	// parse every function file contents into a function template object
+	for functionName, ffc := range functionsFileContents {
+		functionTemplate, err := zftf.createFunctionTemplate(*ffc, functionName)
+		if err != nil {
+			zftf.logger.WarnWith("Failed to create function template",
+				"functionName",
+				functionName,
+				"err",
+				err)
+		}
+
+		functionTemplates = append(functionTemplates, functionTemplate)
+	}
+
+	return functionTemplates, nil
+}
+
+func (zftf *ZipFunctionTemplateFetcher) readZipFile(zf *zip.File) ([]byte, error) {
+	f, err := zf.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	return ioutil.ReadAll(f)
+}
+
+func (zftf *ZipFunctionTemplateFetcher) getZipFileContents(zf *zip.File) (string, error) {
+	unzippedFileBytes, err := zftf.readZipFile(zf)
+	if err != nil {
+		return "", err
+	}
+
+	return string(unzippedFileBytes), nil
+}
+
+func (zftf *ZipFunctionTemplateFetcher) parseFiles(zipReader *zip.Reader) map[string]*FunctionTemplateFileContents {
+	functionTemplateFileContents := make(map[string]*FunctionTemplateFileContents)
+
+	// Read all the files from zip archive
+	for _, zipFile := range zipReader.File {
+		fileName := zipFile.Name
+		splitFileName := strings.Split(fileName, "/")
+
+		// make sure it's a file inside a valid function folder path and it's not a folder
+		if zipFile.FileInfo().IsDir() || len(splitFileName) != 3 {
+			continue
+		}
+
+		functionName := splitFileName[1]
+		fileNameWithoutPath := splitFileName[2]
+
+		// get file contents
+		fileContents, err:= zftf.getZipFileContents(zipFile)
+		if err != nil {
+			zftf.logger.WarnWith("Failed to get zip file contents. Ignoring file", "fileName", fileName, "err", err)
+			continue
+		}
+
+		// if functionTemplateFileContents was not created for this function - create it
+		if _, ok := functionTemplateFileContents[functionName]; !ok {
+			zftf.logger.Debug("Found new function files", "functionName", functionName)
+
+			functionTemplateFileContents[functionName] = &FunctionTemplateFileContents{}
+		}
+		fs := functionTemplateFileContents[functionName]
+
+		if fileNameWithoutPath == "function.yaml" {
+			fs.TemplateAndValues = fileContents
+
+		} else if len(fileNameWithoutPath) > 3 && fileNameWithoutPath[len(fileNameWithoutPath)-3:] == ".py" {
```

Comment:
how about go, js, etc?
also just check if string contains `.values`, `.templates`, not with confusing indices